### PR TITLE
chore: updating the user profile in mixpanel after user login

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
this PR updates the user profile in mixpanel after user login

image

image

## Motivation and Context
this PR distinctly identifies all the users even logging in from a single device based on the email

## How did you test it?
tested locally on mixpanel, users are logged distinctly using their email

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Comment by gitanjli525** [comment]
Updated the PR so as to distinctly log users based on their email,
image

Distinct id set as email
Email being set in user properties
All user events logging under that email itself
image

File: src/entryPoints/AuthModule/Common/CommonAuthUtils.res [hunk 1]
```diff
@@ -183,3 +183,16 @@ module ToggleLiveTestMode = {
...
+
+let onUserLogin = email => {
+  if email->LogicUtils.isNonEmptyString {
+    let mixpanelUserInfo =
+      [("email", email->JSON.Encode.string)]
+      ->Dict.fromArray
+      ->JSON.Encode.object
+    let userId = email
+    LocalStorage.setItem("deviceid", userId)
+    MixPanel.identify(userId)
```
Comment (PritishBudhiraja) [inline src/entryPoints/AuthModule/Common/CommonAuthUtils.res:195]: directly put email why separate id is there?

File: src/entryPoints/AuthModule/Common/CommonAuthUtils.res [hunk 2]
```diff
@@ -183,3 +183,16 @@ module ToggleLiveTestMode = {
     </>
   }
 }
+
+let onUserLogin = email => {
+  if email->LogicUtils.isNonEmptyString {
+    let mixpanelUserInfo =
+      [("email", email->JSON.Encode.string)]
+      ->Dict.fromArray
+      ->JSON.Encode.object
```
Comment (PritishBudhiraja) [inline src/entryPoints/AuthModule/Common/CommonAuthUtils.res:192]: some logicutils fun is there

Comment (PritishBudhiraja) [reply src/entryPoints/AuthModule/Common/CommonAuthUtils.res:192]: In this whole file please change this as it was might have missed in this file

File: src/entryPoints/HyperSwitchApp.res [hunk 3]
```diff
@@ -18,6 +18,7 @@ let make = () => {
     setQuickStartPageState,
     isProdIntentCompleted,
   } = React.useContext(GlobalProvider.defaultContext)
+
```
Comment (PritishBudhiraja) [inline src/entryPoints/HyperSwitchApp.res:21]: remove this unnecessary file commit

File: src/entryPoints/AuthModule/Common/CommonAuthUtils.res [hunk 4]
```diff
@@ -183,3 +183,16 @@ module ToggleLiveTestMode = {
...
+    let userId = email
+    LocalStorage.setItem("deviceid", userId)
```
Comment (PritishBudhiraja) [inline src/entryPoints/AuthModule/Common/CommonAuthUtils.res:194]: Make it as camelCase - deviceId and change in MixpanelHook.res file too

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by PritishBudhiraja — DISMISSED** [review]
LGTM

**Review by sagarnaikjuspay — DISMISSED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Review by Riddhiagrawal001 — DISMISSED** [review]

File: src/entryPoints/AuthModule/Common/CommonAuthUtils.res [hunk 5]
```diff
@@ -183,3 +183,12 @@ module ToggleLiveTestMode = {
     </>
   }
 }
+
+let setUserInMixpanel = email => {
```
Comment (JeevaRamu0104) [inline src/entryPoints/AuthModule/Common/CommonAuthUtils.res:187]: wrap this function with try catch block

**Review by PritishBudhiraja — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**